src/Microsoft.DotNet.Archive/ConsoleProgressReport.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.DotNet.Archive
+{
+    public class ConsoleProgressReport : IProgress<ProgressReport>
+    {
+        string currentPhase;
+        int lastLineLength = 0;
+        double lastProgress = -1;
+        Stopwatch stopwatch;
+
+        public void Report(ProgressReport value)
+        {
+            long progress = (long)(100 * ((double)value.Ticks / value.Total));
+
+            if (progress == lastProgress && value.Phase == currentPhase)
+            {
+                return;
+            }
+            lastProgress = progress;
+
+            lock (this)
```

Comment:
It isn't terrible since this object is completely internal, but I can safeguard and use a lock object.